cmd/archive_upload_client/client.go

```diff
@@ -12,23 +12,48 @@ import (
 	"log"
 	"os"
 	"strings"
-	"time"
 
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
 	"google.golang.org/api/idtoken"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/oauth"
 
 	pb "github.com/routeviews/google-cloud-storage/proto/rv"
-	grpcMetadata "google.golang.org/grpc/metadata"
 )
 
 var (
-	server = flag.String("server", "localhost:9876", "The host:port of the gRPC server.")
-	file   = flag.String("file", "", "A local File to transfer to cloud storage.")
+	server         = flag.String("server", "localhost:9876", "The host:port of the gRPC server.")
+	file           = flag.String("file", "", "A local File to transfer to cloud storage.")
+	serviceAccount = flag.String("sa_key", "", "Service account private key.")
```

Comment:
you might consider renaming this from serviceAccount so saKey ?
since you are referring to the key not the user.